src/mame/drivers/rf51w14cg.cpp

```diff
@@ -1,138 +1,138 @@
+
+ JPM7 = CLK/128 / CLK/64
+
+*******************************************************************************/
+
+#include "emu.h"
+
+#include "cpu/i86/i186.h"
+#include "sound/okim6376.h"
+
+#include "speaker.h"
+
+namespace
+{
+
+class rf51w14cg_state : public driver_device
+{
+public:
+	rf51w14cg_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_maincpu(*this, "maincpu")
+		, m_okim6376(*this, "oki")
+	{
+	}
+
+	void rf51w14cg(machine_config &config);
+
+protected:
+	required_device <cpu_device> m_maincpu;
+	required_device <okim6376_device> m_okim6376;
+};
+
+static INPUT_PORTS_START( rf51w14cg )
+	PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+
+	PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+
+	PORT_START("DSW3")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+INPUT_PORTS_END
+
+void rf51w14cg_state::rf51w14cg(machine_config &config)
+{
+	I80188(config, m_maincpu, 32_MHz_XTAL / 2); // Intel N80C188XL-20, guess divisor
+
+	SPEAKER(config, "mono").front_center();
+
+	OKIM6376(config, m_okim6376, XTAL(9'830'400)/64).add_route(ALL_OUTPUTS, "mono", 1.0); // Frecuency divisor as per JMP7
+}
+
+// The board was found with the program ROMs sockets unpopulated and the M48T08 with the battery dead
+ROM_START( rf51w14cg )
+	ROM_REGION( 0x100000, "maincpu", 0 )
+	ROM_LOAD( "high.bin", 0x00000, 0x80000, NO_DUMP )
+	ROM_LOAD( "low.bin",  0x80000, 0x80000, NO_DUMP )
+
+	ROM_REGION( 0x100000, "oki", 0 )
+	ROM_LOAD( "recreativos_franco_sonido-1_321dabf_01083c.u3", 0x00000, 0x80000, CRC(bd9bb391) SHA1(f08da81544e6b8c518634d081bf68d862b90c099) )
+	ROM_LOAD( "recreativos_franco_sonido-2_7c8bced_01083d.u4", 0x80000, 0x80000, CRC(cf8e7957) SHA1(5d30d7f15c1690b819e467fc308f12f97577b906) )
+
+	ROM_REGION( 0x2000, "nvram", 0 )
+	ROM_LOAD( "recreativos_franco_mod_rf51w14-cg_2003_m48t08.bin", 0x0000, 0x2000, NO_DUMP )
+ROM_END
+
+} // anonymous namespace
+
+//    YEAR  NAME       PARENT  MACHINE    INPUT      CLASS            INIT        ROT   COMPANY               FULLNAME                                                      FLAGS
+GAME( 2003, rf51w14cg, 0,      rf51w14cg, rf51w14cg, rf51w14cg_state, empty_init, ROT0, "Recreativos Franco", "unknown Recreativos Franco slot machine (model RF51W14-CG)", MACHINE_IS_SKELETON_MECHANICAL )
```

Comment:
What have you done with this file?

Reply:
Nothing. I don't know why this file appears in this pull-request. Carriage return issue ?